### PR TITLE
Fix direct running mode multi_processing on win32

### LINKED ISSUE #20427: tests fails on windows - PortableRunnerTestWithSubprocesses causes grpc Deadline Exceeded
apache_beam.runners.portability.portable_runner_test.PortableRunnerTestWithSubprocesses causes grpc Deadline Exceeded on windows

Link to the github workflow run with mentioned error:
[https://github.com/TobKed/beam/runs/937577885?check_suite_focus=true](https://github.com/TobKed/beam/runs/937577885?check_suite_focus=true)

partial log:

FsFException in thread wait_until_finish_read:

Traceback (most recent call last):

File "c:\hostedtoolcache\windows\python\3.6.8\x64\lib\threading.py", line 916, in _bootstrap_inner

self.run()

File "c:\hostedtoolcache\windows\python\3.6.8\x64\lib\threading.py", line 864, in run

self._target(*self._args, **self._kwargs)

File "D:\a\beam\beam\sdks\python\apache_beam\runners\portability\portable_runner.py", line 512, in read_messages

for message in self._message_stream:

File "d:\a\beam\beam\sdks\python\target\.tox\py36-win\lib\site-packages\grpc\_channel.py", line 416, in __next__

return self._next()

File "d:\a\beam\beam\sdks\python\target\.tox\py36-win\lib\site-packages\grpc\_channel.py", line 706, in _next

raise self

grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:

status = StatusCode.DEADLINE_EXCEEDED

details = "Deadline Exceeded"

debug_error_string = "\{"created":"@1596375957.859000000","description":"Error received from peer ipv6:[::1]:50984","file":"src/core/lib/surface/call.cc","file_line":1055,"grpc_message":"Deadline Exceeded","grpc_status":4}

Imported from Jira [BEAM-10625](https://issues.apache.org/jira/browse/BEAM-10625). Original Jira may contain additional context.
Reported by: TobKed.

### LINKED ISSUE #22039: [Bug]: unable to set up multiprocessing
### What happened?

Beam - 2.39.0
Python - 3.8.13
Win10
Runner - Direct

I was following this page for setting some parrallelism (threading in this case):
https://beam.apache.org/documentation/runners/direct/

However, regardless of how I pass the required parameters to PipelineOptions instance, I get the error below. This error is really obscure, and I have no idea how to fix it (as you might've guessed, googling didn't help much).
Mind you, I have no issues when those parameters are set to defaults.

```
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function annotate_downstream_side_inputs at 0x000001B6F91B8AF0> ====================
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function fix_side_input_pcoll_coders at 0x000001B6F91B8C10> ====================
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function pack_combiners at 0x000001B6F91B9160> ====================
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function lift_combiners at 0x000001B6F91B91F0> ====================
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function expand_sdf at 0x000001B6F91B93A0> ====================
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function expand_gbk at 0x000001B6F91B9430> ====================
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function sink_flattens at 0x000001B6F91B9550> ====================
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function greedily_fuse at 0x000001B6F91B95E0> ====================
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function read_to_impulse at 0x000001B6F91B9670> ====================
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function impulse_to_input at 0x000001B6F91B9700> ====================
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function sort_stages at 0x000001B6F91B9940> ====================
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function add_impulse_to_dangling_transforms at 0x000001B6F91B9A60> ====================
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function setup_timer_mapping at 0x000001B6F91B98B0> ====================
INFO:apache_beam.runners.portability.fn_api_runner.translations:==================== <function populate_data_channel_coders at 0x000001B6F91B99D0> ====================
INFO:apache_beam.runners.portability.fn_api_runner.worker_handlers:starting control server on port 49986
INFO:apache_beam.runners.portability.fn_api_runner.worker_handlers:starting data server on port 49987
INFO:apache_beam.runners.portability.fn_api_runner.worker_handlers:starting state server on port 49988
INFO:apache_beam.runners.portability.fn_api_runner.worker_handlers:starting logging server on port 49989
INFO:apache_beam.runners.portability.fn_api_runner.worker_handlers:Created Worker handler <apache_beam.runners.portability.fn_api_runner.worker_handlers.EmbeddedGrpcWorkerHandler object at 0x000001B6FB280820> for environment ref_Environment_default_environment_1 (beam:env:embedded_python_grpc:v1, b'{}')
INFO:apache_beam.runners.worker.statecache:Creating state cache with size 100
INFO:apache_beam.runners.worker.sdk_worker:Creating insecure control channel for localhost:49986.
INFO:apache_beam.runners.worker.sdk_worker:Control channel established.
INFO:apache_beam.runners.worker.sdk_worker:Initializing SDKHarness with unbounded number of workers.
INFO:apache_beam.runners.portability.fn_api_runner.worker_handlers:Created Worker handler <apache_beam.runners.portability.fn_api_runner.worker_handlers.EmbeddedGrpcWorkerHandler object at 0x000001B6FBCE0D90> for environment ref_Environment_default_environment_1 (beam:env:embedded_python_grpc:v1, b'{}')
INFO:apache_beam.runners.worker.statecache:Creating state cache with size 100
INFO:apache_beam.runners.worker.sdk_worker:Creating insecure control channel for localhost:49986.
INFO:apache_beam.runners.worker.sdk_worker:Control channel established.
INFO:apache_beam.runners.worker.sdk_worker:Initializing SDKHarness with unbounded number of workers.
INFO:apache_beam.runners.worker.sdk_worker:Creating insecure state channel for localhost:49988.
INFO:apache_beam.runners.worker.sdk_worker:State channel established.
INFO:apache_beam.runners.worker.data_plane:Creating client data channel for localhost:49987
INFO:apache_beam.runners.worker.sdk_worker:Creating insecure state channel for localhost:49988.
INFO:apache_beam.runners.worker.sdk_worker:State channel established.
INFO:apache_beam.runners.worker.data_plane:Creating client data channel for localhost:49987
INFO:apache_beam.runners.worker.sdk_worker:No more requests from control plane
INFO:apache_beam.runners.worker.sdk_worker:SDK Harness waiting for in-flight requests to complete
INFO:apache_beam.runners.worker.data_plane:Closing all cached grpc data channels.
INFO:apache_beam.runners.worker.sdk_worker:Closing all cached gRPC state handlers.
INFO:apache_beam.runners.worker.sdk_worker:Done consuming work.
INFO:apache_beam.runners.worker.sdk_worker:No more requests from control plane
INFO:apache_beam.runners.worker.sdk_worker:SDK Harness waiting for in-flight requests to complete
INFO:apache_beam.runners.worker.data_plane:Closing all cached grpc data channels.
INFO:apache_beam.runners.worker.sdk_worker:Closing all cached gRPC state handlers.
INFO:apache_beam.runners.worker.sdk_worker:Done consuming work.
Traceback (most recent call last):
  File "T:/nvme1/akutergin/OCR/beam_pipelines/main.py", line 132, in <module>
    ocrd = p_markup.run(decoded, models_and_libs, templates)
  File "C:\Anaconda3\envs\ML-5\lib\site-packages\apache_beam\pipeline.py", line 596, in __exit__
    self.result = self.run()
  File "C:\Anaconda3\envs\ML-5\lib\site-packages\apache_beam\pipeline.py", line 573, in run
    return self.runner.run_pipeline(self, self._options)
  File "C:\Anaconda3\envs\ML-5\lib\site-packages\apache_beam\runners\direct\direct_runner.py", line 131, in run_pipeline
    return runner.run_pipeline(pipeline, options)
  File "C:\Anaconda3\envs\ML-5\lib\site-packages\apache_beam\runners\portability\fn_api_runner\fn_runner.py", line 199, in run_pipeline
    self._latest_run_result = self.run_via_runner_api(
  File "C:\Anaconda3\envs\ML-5\lib\site-packages\apache_beam\runners\portability\fn_api_runner\fn_runner.py", line 208, in run_via_runner_api
    return self.run_stages(stage_context, stages)
  File "C:\Anaconda3\envs\ML-5\lib\site-packages\apache_beam\runners\portability\fn_api_runner\fn_runner.py", line 408, in run_stages
    bundle_results = self._execute_bundle(
  File "C:\Anaconda3\envs\ML-5\lib\site-packages\apache_beam\runners\portability\fn_api_runner\fn_runner.py", line 736, in _execute_bundle
    self._run_bundle(
  File "C:\Anaconda3\envs\ML-5\lib\site-packages\apache_beam\runners\portability\fn_api_runner\fn_runner.py", line 965, in _run_bundle
    result, splits = bundle_manager.process_bundle(
  File "C:\Anaconda3\envs\ML-5\lib\site-packages\apache_beam\runners\portability\fn_api_runner\fn_runner.py", line 1388, in process_bundle
    process_bundle=beam_fn_api_pb2.ProcessBundleResponse(
  File "C:\Anaconda3\envs\ML-5\lib\site-packages\apache_beam\metrics\monitoring_infos.py", line 421, in consolidate
    yield reduce(merge, values)
  File "C:\Anaconda3\envs\ML-5\lib\site-packages\apache_beam\metrics\monitoring_infos.py", line 419, in merge
    payload=combiner(a.payload, b.payload))
  File "C:\Anaconda3\envs\ML-5\lib\site-packages\apache_beam\metrics\monitoring_infos.py", line 384, in distribution_payload_combiner
    (count_a, sum_a, min_a, max_a) = _decode_distribution(coder, payload_a)
  File "C:\Anaconda3\envs\ML-5\lib\site-packages\apache_beam\metrics\monitoring_infos.py", line 451, in _decode_distribution
    value_coder.decode_from_stream(stream, True),
  File "apache_beam\coders\coder_impl.py", line 914, in apache_beam.coders.coder_impl.VarIntCoderImpl.decode_from_stream
  File "apache_beam\coders\coder_impl.py", line 916, in apache_beam.coders.coder_impl.VarIntCoderImpl.decode_from_stream
  File "apache_beam\coders\stream.pyx", line 195, in apache_beam.coders.stream.InputStream.read_var_int64
RuntimeError: VarLong too long.
```

### Issue Priority

Priority: 0

### Issue Component

Component: runner-direct

### PR DESCRIPTION
**Pull request by Abacn**
Fixes #20427
Fixes #22039

Currently the subprocess worker make calls of `subprocess.Popen` with arguments of bytes type (received from grpc). This will cause error in win32: 
`TypeError: bytes args is not allowed on Windows`

This PR encodes the argument passed to subprocess.Popen to str.

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Abacn** [comment]
R: @tvalentyn

File: sdks/python/apache_beam/runners/portability/local_job_service.py [hunk 1]
```diff
@@ -184,7 +185,11 @@ def __init__(
       control_address,
       provision_info,
       worker_id=None):
-    self._worker_command_line = worker_command_line
+    if sys.platform == 'win32' and isinstance(worker_command_line, bytes):
+      # Fix subprocess.Popen not support bytes args
+      self._worker_command_line = worker_command_line.decode()
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/runners/portability/local_job_service.py:190]: is it possible that decoding will work both on windows and other systems? If so then maybe just always call decode.

**Review by ryanthompson591 — COMMENTED** [review]
I suggest describing what this PR does and why in the description, instead of just fixes issue XXXX.

File: sdks/python/apache_beam/runners/portability/local_job_service.py (hunk 1 above)
Comment (Abacn) [reply sdks/python/apache_beam/runners/portability/local_job_service.py:190]: I believe so, inside the sdk the bytes are encoded here with utf-8:
https://github.com/apache/beam/blob/60581e8b1b6e93889cce78542e99d1fea4105d54/sdks/python/apache_beam/transforms/environments.py#L729
And decode utf-8 bytes should be safe as long as user not call it from elsewhere (e.g. a non utf-8 locale environment and run the subprocess worker manually, which should not be a normal use case).

**Review by tvalentyn — APPROVED** [review]

**Pull request merged**